### PR TITLE
Add CareerInfo and merge sign up flow

### PR DESCRIPTION
**Pull request by heartjigi**
Add CareerInfo and merge sign up flow
General Info -> Career Info -> Personality.

I'll add server communication on Career Info on adding  server refactoring

@ethkim @ace0625 @MechanicKim @ifndefdeadmau5 @SarahhhC

### CODE REVIEW

File: ios/Bridgeme.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -821,8 +874,8 @@
 					/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include,
 					"$(SRCROOT)/../node_modules/react-native/React/**",
 					"$(SRCROOT)/../node_modules/react-native-fbsdk/ios/RCTFBSDK/**",
-					"$(SRCROOT)\..\node_modules\react-native-linear-gradient\BVLinearGradient",
-					"$(SRCROOT)\..\node_modules\react-native-image-picker\ios/**",
+					"$(SRCROOT)..\node_modules\neact-native-linear-gradientBVLinearGradient",
```
Comment (SarahhhC) [inline ios/Bridgeme.xcodeproj/project.pbxproj:902]: Is this neact? not react?

File: src/components/SignUp/CareerMETA.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    '4-6 years',
+    '6+ years',
+  ],
+  education_background: [
+    "Associate's",
+    "Bachelor's",
+    "Master's",
+    'PhD',
+    'Other',
+  ],
```
Comment (SarahhhC) [inline src/components/SignUp/CareerMETA.js]: Does front get career information from CareerMETA? not server?

Comment (heartjigi) [reply src/components/SignUp/CareerMETA.js]: Yeah we will get that data on front....

File: ios/Bridgeme.xcodeproj/project.pbxproj (hunk 1 above)
Comment (heartjigi) [reply ios/Bridgeme.xcodeproj/project.pbxproj:902]: maybe this made by rnpm I exactly don't know why is this

File: src/components/App.js [hunk 3]
```diff
@@ -2,6 +2,9 @@ import React, { Component } from 'react';
 import { Platform, StyleSheet, Alert } from 'react-native';
 import Login from './Login';
```
Comment (ace0625) [inline src/components/App.js]: sort

File: src/components/Login.js [hunk 4]
```diff
@@ -71,6 +71,7 @@ class Login extends Component {
   }
 
   onLoginSuccess(result) {
+
```
Comment (ace0625) [inline src/components/Login.js]: delete the empty space

File: src/components/SignUp/CareerInfo.js [hunk 5]
```diff
@@ -0,0 +1,255 @@
+import React, { Component } from 'react';
+import {
+  Image,
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js:5]: sort

**Comment by heartjigi** [comment]
lgtm

File: src/components/SignUp/CareerInfo.js [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+    super(props);
+    this.state = {
+      questions: [
+        'What area do you work in?',
+        'What roles do you work in?',
+        'Years of work experience',
+        'Last Educational Background?',
+      ],
+      selected: ['', '', '', ''],
+      option: ['', '', '', ''],
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js]: ??

File: src/components/SignUp/CareerInfo.js [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+    if (Platform.OS === 'ios') {
+      for (idx = 0; idx < this.state.questions.length; idx++) {
+        updatePosition(this.refs['SELECT' + idx]);
+        updatePosition(this.refs['OPTION' + idx]);
+      }
+    }
+
+  }
+
+  _getOptionList(index) {
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js]: no underscore

File: src/components/SignUp/CareerInfo.js [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+    this.state.selectOP = index;
+    this.forceUpdate();
+    return this.refs['OPTION' + String(index)];
+  }
+
+  _onSelect(province, idx) {
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js]: no underscore

File: src/components/SignUp/CareerInfo.js [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+          this.state.clearFlag = true;
+          break;
+        }
+      }
+    }
+
+    this.forceUpdate();
+  }
+
+  _getOptionSet(index) {
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js]: no underscore

File: src/components/SignUp/CareerInfo.js [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+    return this.state.option[index].map(
+      (area, idx) => <Option key={idx}>{area}</Option>
+    );
+
+  }
+
+  _getQuestionSet() {
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js]: no underscore

File: src/components/SignUp/CareerInfo.js [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+        }
+      );
+  }
+
+  onUploadCallback() {
+    Actions.expertInfo({ me: this.props.me });
+  }
+
+  render() {
+
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js:188]: No empty space

File: src/components/SignUp/CareerInfo.js [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+                  <Dropdown
+                    style={{ height: 40, width: Dimensions.get('window').width - 60 }}
+                    values={this.state.option[idx]}
+                    onChange={(data) => {
+                      this._onSelect(data.value, idx);
+                    }}/>
+
+                  </View>
+              </View>
+
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js]: delete the empty space

File: src/components/SignUp/CareerInfo.js [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+                colors={['#07e4dd', '#44acff']}>
+                <Text style={styles.buttonText}>NEXT</Text>
+              </LinearGradient>
+            </TouchableOpacity>
+          </View>
+        </ScrollView>
+      </View>
+    );
+  }
+
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js]: delete the empty space

File: src/components/SignUp/CareerInfo.js [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+      option: ['', '', '', ''],
+      selectOP: '',
+      clearFlag: false,
+    };
+
+    this.state.option[0] = data.area;
+    this.state.option[1] = [];
+    this.state.option[2] = data.years;
+    this.state.option[3] = data.education_background;
+
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js:43]: no empty line

File: src/components/SignUp/CareerInfo.js [hunk 15]
```diff
@@ -0,0 +1,255 @@
...
+  updatePosition
+} from '../../utils/dropdown/index';
+import Dropdown from 'react-native-dropdown-android';
+import LinearGradient from 'react-native-linear-gradient';
+import Progress from '../Shared/Progress';
+import { Actions, Scene, }  from 'react-native-router-flux';
+import data from './CareerMETA';
+
+class CareerInfo extends Component {
+
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js]: no empty line

File: src/components/SignUp/CareerInfo.js [hunk 16]
```diff
@@ -0,0 +1,255 @@
...
+  }
+
+  componentDidMount() {
+    if (Platform.OS === 'ios') {
+      for (idx = 0; idx < this.state.questions.length; idx++) {
+        updatePosition(this.refs['SELECT' + idx]);
+        updatePosition(this.refs['OPTION' + idx]);
+      }
+    }
+
```
Comment (ace0625) [inline src/components/SignUp/CareerInfo.js]: no empty line

File: src/components/Login.js (hunk 4 above)
Comment (heartjigi) [reply src/components/Login.js]: solved

File: src/components/SignUp/CareerInfo.js (hunk 5 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js:5]: solved

File: src/components/SignUp/CareerInfo.js (hunk 15 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: solved

File: src/components/SignUp/CareerInfo.js (hunk 6 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: for dynamic data change with first answer

File: src/components/SignUp/CareerInfo.js (hunk 14 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js:43]: solved

File: src/components/SignUp/CareerInfo.js (hunk 16 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: solved

File: src/components/SignUp/CareerInfo.js (hunk 7 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: solved

File: src/components/SignUp/CareerInfo.js (hunk 8 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: solved

File: src/components/SignUp/CareerInfo.js (hunk 9 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: solved

File: src/components/SignUp/CareerInfo.js (hunk 10 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: solved

File: src/components/SignUp/CareerInfo.js (hunk 12 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: solved

File: src/components/SignUp/CareerInfo.js (hunk 11 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js:188]: solved

File: src/components/SignUp/CareerInfo.js (hunk 13 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: solved

File: src/components/App.js (hunk 3 above)
Comment (heartjigi) [reply src/components/App.js]: solved

File: src/components/SignUp/CareerInfo.js (hunk 6 above)
Comment (ace0625) [reply src/components/SignUp/CareerInfo.js]: That's not how you use arrays in Javascript

Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: Chaged!

File: android/Dropdown/src/main/java/com/chymtt/reactnativedropdown/DropdownPackage.java [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+        return Arrays.<ViewManager>asList(
+                new DropdownManager()
+        );
+    }
+
+    @Override
+    public List<Class<? extends JavaScriptModule>> createJSModules() {
+        return Arrays.asList();
+    }
+}
```
Comment (SarahhhC) [inline android/Dropdown/src/main/java/com/chymtt/reactnativedropdown/DropdownPackage.java]: Add new line.

File: src/components/SignUp/ExpertInfo.js [hunk 18]
```diff
@@ -0,0 +1,175 @@
...
+    console.log(this.state.checked);
+  }
+
+  // Update checkbox state
+  updateCheckBox(answerIdx, optionIdx, isFreeForm, isChecked) {
+    this.state.checked[optionIdx] = !this.state.checked[optionIdx];
+    this.forceUpdate();
+  }
+
+  _getOptionSet() {
```
Comment (SarahhhC) [inline src/components/SignUp/ExpertInfo.js]: Delete _ in function name~

File: src/components/SignUp/GeneralInfo.js [hunk 19]
```diff
@@ -166,8 +168,15 @@ class GeneralInfo extends Component {
...
+  onUploadCallback(result, error) {
+    if (result)
+      Actions.main({ me: this.props.me });
+    if (error)
+      alert(JSON.stringify(error));
+  }
+
   onUploadSuccess(result) {
-    Actions.main({ me: this.props.me });
+    Actions.careerInfo({ me: this.props.me });//Actions.main({ me: this.props.me });
```
Comment (SarahhhC) [inline src/components/SignUp/GeneralInfo.js]: Why did you remain comments?
If you never use it, delete it!

File: src/utils/CheckBox.js [hunk 20]
```diff
@@ -0,0 +1,91 @@
...
+    justifyContent: 'center',
+  },
+  label: {
+    marginLeft: 10,
+    fontSize: 14,
+    color: '#2e3031',
+  },
+});
+
+const checkBox = new CheckBox;
```
Comment (SarahhhC) [inline src/utils/CheckBox.js:90]: Where do you use variable checkBox??

File: src/utils/dropdown/optionList.js [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+  currentPosition(pageX, pageY) {
+    this.setState({
+      ...this.state,
+      pageX,
+      pageY,
+    });
+  }
+
+  show(items, selected, positionX, positionY, width, height, onSelect) {
+    let h = (items.length > 3) ? height * 3 + height / 4 : height * (items.length + 1);
```
Comment (SarahhhC) [inline src/utils/dropdown/optionList.js:39]: What does number 3 or 4 mean??

File: src/components/SignUp/CareerInfo.js [hunk 22]
```diff
@@ -0,0 +1,246 @@
...
+      selected: [],
+      option: [],
+      selectOP: '',
+      clearFlag: false,
+    };
+
+    this.state.option[0] = data.area;
+    this.state.option[1] = [];
+    this.state.option[2] = data.years;
+    this.state.option[3] = data.education_background;
```
Comment (ochanje210) [inline src/components/SignUp/CareerInfo.js]: does this work without using setState?

File: src/components/SignUp/GeneralInfo.js (hunk 19 above)
Comment (heartjigi) [reply src/components/SignUp/GeneralInfo.js]: solved

File: src/components/SignUp/CareerInfo.js (hunk 22 above)
Comment (heartjigi) [reply src/components/SignUp/CareerInfo.js]: setState just for re-render, and at this spot doen't need re-render.

File: src/utils/dropdown/optionList.js (hunk 21 above)
Comment (heartjigi) [reply src/utils/dropdown/optionList.js:39]: 3 means showing line, and 4 is remain of next option for easy to recognize scrollable

File: src/utils/CheckBox.js (hunk 20 above)
Comment (heartjigi) [reply src/utils/CheckBox.js:90]: for using without allocation

**Comment by lyw0149** [comment]
Thanks Dabin!! LGTM

**Comment by lyw0149** [comment]
LGTM

**Pull request merged**